compiler/damlc/tests/daml-test-files/ComposedKey.daml

```diff
@@ -12,7 +12,16 @@ module ComposedKey where
 
 import DA.Assert
 import DA.Text
-import GenTemplCompat
+
+template Fact with
```

Comment:
It would be nice to have a comment here on which part of this test triggers the bug that you fixed.

Reply:
I cannot pin down which part triggers the bug. I think it's a combination of things. But the instant inlining of super class dictionaries GHC does seems reasonable. I'm not too fussed about this TBH.